lib/http.c

```diff
@@ -2692,7 +2694,8 @@ CURLcode Curl_http(struct Curl_easy *data, bool *done)
     }
     else
 #endif
-      DEBUGASSERT(Curl_conn_http_version(data, conn) == 20);
+    DEBUGASSERT(Curl_conn_http_version(data, conn) == 20);
```

Comment:
Hm, the line belongs to the else branch, depending on config. I wonder if there is a way to make this more clear. Perhaps dulicate the line to make it more explicit?

Reply:
Not sure myself how this is supposed to work/what is exactly checked. But that is outside of what this PR changes. I'd like to keep this out of here.